File: slack-reader/src/test/java/dev/number6/slackreader/dagger/FakeClockModule.java
package dev.number6.slackreader.dagger;

import dev.number6.slackreader.generate.SlackReaderRDG;
import dagger.Module;
import dagger.Provides;
import uk.org.fyodor.range.Range;

import javax.inject.Singleton;
import java.time.Clock;
import java.time.LocalDateTime;
import java.time.ZoneId;
import java.time.ZoneOffset;

@Module
public class FakeClockModule {

    @Provides
    @Singleton
    Clock providesClock() {
        return Clock.fixed(LocalDateTime.now().minusDays(SlackReaderRDG.longVal(Range.closed(2L, 10L)).next()).toInstant(ZoneOffset.UTC), ZoneId.systemDefault());
    }
}

File: slack-reader/src/test/java/dev/number6/slackreader/dagger/FakeClockModule.kt
package dev.number6.slackreader.dagger

import dagger.Module
import dagger.Provides
import uk.org.fyodor.generators.RDG
import uk.org.fyodor.range.Range
import java.time.Clock
import java.time.LocalDateTime
import java.time.ZoneId
import java.time.ZoneOffset
import javax.inject.Singleton

@Module
class FakeClockModule {
    @Provides
    @Singleton
    fun providesClock(): Clock {
        return Clock.fixed(LocalDateTime.now().minusDays(RDG.longVal(Range.closed(2L, 10L)).next()).toInstant(ZoneOffset.UTC), ZoneId.systemDefault())
    }
}